met-api/src/met_api/services/comment_service.py

```diff
@@ -158,32 +161,90 @@ def extract_comments_from_survey(cls, survey_submission: SubmissionSchema, surve
     @classmethod
     def export_comments_to_spread_sheet(cls, survey_id):
         """Export comments to spread sheet."""
-        engagement = SurveyModel.find_by_id(survey_id)
-        one_of_roles = (
-            MembershipType.TEAM_MEMBER.name,
-            Role.EXPORT_TO_CSV.value
-        )
-        authorization.check_auth(one_of_roles=one_of_roles, engagement_id=engagement.engagement_id)
-        comments = Comment.get_comments_by_survey_id(survey_id)
-        formatted_comments = [
-            {
-                'commentNumber': comment.id,
-                'dateSubmitted': str(comment.submission_date),
-                'author': '',
-                'commentText': comment.text,
-                'reviewer': comment.reviewed_by,
-                'exportDate': str(datetime.utcnow())
-            }
-            for comment in comments]
-
-        data = {
-            'comments': formatted_comments
+        try:
+            engagement = SurveyModel.find_by_id(survey_id)
+            one_of_roles = (
+                MembershipType.TEAM_MEMBER.name,
+                Role.EXPORT_TO_CSV.value
+            )
+            authorization.check_auth(one_of_roles=one_of_roles, engagement_id=engagement.engagement_id)
+            comments = Comment.get_comments_by_survey_id(survey_id)
+            metadata_model = EngagementMetadataModel.find_by_id(engagement.engagement_id)
+            project_name = metadata_model.project_metadata.get('project_name') if metadata_model else None
+        except BusinessException as err:
```

Comment:
should this try catch be here? Wouldn't the one in the resource be able to handle it

Reply:
yes, removed it.